### ISSUE TITLE
Any Error When Sending Message With Expired Access Token

### ISSUE THREAD

**Issue by 1337-sys** [chatwoot/chatwoot#9708]
### Describe the bug

Hi ChatWoot Team,  

Did the message actually go out successfully?

I am trying WhatsApp cloud inbox via chatwoot, but the message is not sent because the token is expired, but on the chatwoot screen, the message seems to have been sent successfully. In the messages table, there does not appear to be a message_id starting with "mid.***" in the source_id column.

Logs: 

```
chatwoot-sidekiq-1   | E, [2024-07-01T18:06:30.244696 #1] ERROR -- : [ActiveJob] [SendReplyJob] [530d1adb-e119-4a24-b4af-b4b4be1f008f] {"error":{"message":"Error validating access token: Session has expired on Saturday, 29-Jun-24 09:00:00 PDT. The current time is Monday, 01-Jul-24 11:06:30 PDT.","type":"OAuthException","code":190,"error_subcode":463,"fbtrace_id":"AZ-HzGEkqc5Vf3E0gy-eAcV"}}
```

Sincerely :)

### To Reproduce

_

### Environment

Docker

### Cloud Provider

Other [please specify in the description]

### Platform

Browser

### Operating system

Mac OS

**Comment by 1337-sys** [comment]
At least the user should be notified when the message is not sent.

**Comment by 1337-sys** [comment]
A checking code may be required, just like on Instagram:

```
if response[:error].present?
      Rails.logger.error("Instagram response: #{response['error']} : #{message_content}")
      message.status = :failed
      message.external_error = external_error(response)
end
```

**Comment by pranavrajs** [comment]
@1337-sys This is already handled in WhatsApp, please see the segment below. 

https://github.com/chatwoot/chatwoot/blob/5520bf68f37c5bbc189f5c4bbbdc325e1a4fad22/app/services/whatsapp/incoming_message_base_service.rb#L48-L54

1. Are you using an API channel or is it the native WhatsApp inbox? 
2. Have you verified that the webhooks are added properly?

**Comment by 1337-sys** [comment]
Thx for reply @pranavrajs. This only applies to incoming messages. I'm talking about sending messages.

**Comment by pranavrajs** [comment]
@1337-sys No, this is not just incoming messages, the status for the replies are handled here.

**Comment by 1337-sys** [comment]
@pranavrajs  You didn't understand me. What I'm talking about is not status (read, sent, another meta side errors) messages. I can't send the message.. Have you ever looked at the log? It says the access token is expired.

**Comment by sojan-official** [comment]
ref: https://github.com/chatwoot/chatwoot/blob/5520bf68f37c5bbc189f5c4bbbdc325e1a4fad22/app/services/whatsapp/send_on_whatsapp_service.rb#L94

https://github.com/chatwoot/chatwoot/blob/5520bf68f37c5bbc189f5c4bbbdc325e1a4fad22/app/services/whatsapp/providers/whatsapp_cloud_service.rb#L118

It seems like we are only updating the source ID in case of success and not storing the error into the message in case of error. We will need to handle this.

**Comment by 1337-sys** [comment]
@sojan-official Thx. I'm not sure if this is a WhatsApp specific issue, but I can contribute to get it fixed.

**Comment by sojan-official** [comment]
@1337-sys That would be super helpful, thanks a lot!

While it might be beneficial to audit all the channels for this case, for now, let's focus on WhatsApp alone. The current implementation only handles errors when a message is sent and WhatsApp responds with an error status. However, as you mentioned, we're not handling the situation where the initial call to create a message in the WhatsApp endpoint fails. We need to store the error message in this case and mark the message delivery status as failed.

Let us know if you need any help with this!

**Comment by pranavrajs** [comment]
@1337-sys My bad, I was expecting this to be handled on the status callbacks. In fact the message itself is not created at Meta. I didn't realize the difference. 

As @sojan-official suggested we could handle the error case while sending the message.

**Comment by cognitus** [comment]
please fix, I lost a couple hours before figure out 

> As @sojan-official suggested we could handle the error case while sending the message.

it's a good solution

**Comment by sojan-official** [comment]
fixed by https://github.com/chatwoot/chatwoot/pull/11015
